Sources/SecureStorage/SecureVaultFactory.swift

```diff
@@ -57,62 +63,51 @@ public class SecureVaultFactory {
             }
 
             do {
-                let providers = try makeSecureVaultProviders()
-                let vault = DefaultSecureVault(authExpiry: authExpiration, providers: providers)
+                let providers = try makeSecureStorageProviders()
+                let vault = Vault(providers: providers)
 
                 self.vault = vault
 
                 return vault
 
-            } catch let error as SecureVaultError {
+            } catch let error as SecureStorageError {
                 errorReporter?.secureVaultInitFailed(error)
                 throw error
             } catch {
-                errorReporter?.secureVaultInitFailed(SecureVaultError.initFailed(cause: error))
-                throw SecureVaultError.initFailed(cause: error)
+                errorReporter?.secureVaultInitFailed(SecureStorageError.initFailed(cause: error))
+                throw SecureStorageError.initFailed(cause: error)
             }
         }
 
     }
     
-    internal func makeSecureVaultProviders() throws -> SecureVaultProviders {
-        let (cryptoProvider, keystoreProvider): (SecureVaultCryptoProvider, SecureVaultKeyStoreProvider)
+    public func makeSecureStorageProviders() throws -> SecureStorageProviders<Vault.DatabaseProvider> {
+        let (cryptoProvider, keystoreProvider): (SecureStorageCryptoProvider, SecureStorageKeyStoreProvider)
         do {
             (cryptoProvider, keystoreProvider) = try createAndInitializeEncryptionProviders()
         } catch {
-            throw SecureVaultError.initFailed(cause: error)
+            throw SecureStorageError.initFailed(cause: error)
         }
-        guard let existingL1Key = try keystoreProvider.l1Key() else { throw SecureVaultError.noL1Key }
+        guard let existingL1Key = try keystoreProvider.l1Key() else { throw SecureStorageError.noL1Key }
+
+        let databaseProvider: Vault.DatabaseProvider
 
-        let databaseProvider: SecureVaultDatabaseProvider
         do {
 
             do {
-                databaseProvider = try DefaultDatabaseProvider(key: existingL1Key)
-            } catch DefaultDatabaseProvider.DbError.nonRecoverable {
-                databaseProvider = try DefaultDatabaseProvider.recreateDatabase(withKey: existingL1Key)
+                databaseProvider = try self.makeDatabaseProvider(existingL1Key)
+            } catch SecureStorageDatabaseError.nonRecoverable {
+                databaseProvider = try Vault.DatabaseProvider.recreateDatabase(withKey: existingL1Key)
```

Comment:
I know this is old code, but this will result in dataloss.  At a minimum we should have a pixel here.  Maybe one for later though.

Reply:
Yeah I'm happy to jot down a task for this and get a privacy triage up.